src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -625,48 +625,67 @@ private static string Encode5987(StringSegment input)
             ? stackalloc byte[MaxStackAllocSizeBytes]
             : bufferFromPool = ArrayPool<byte>.Shared.Rent(maxInputBytes);
 
+        char[]? charBufferFromPool = null;
+        Span<char> tempCharBuffer = maxInputBytes <= MaxStackAllocSizeBytes
+            ? stackalloc char[MaxStackAllocSizeBytes]
+            : charBufferFromPool = ArrayPool<char>.Shared.Rent(maxInputBytes);
+
         var bytesWritten = Encoding.UTF8.GetBytes(input, inputBytes);
         inputBytes = inputBytes[..bytesWritten];
 
-        int totalBytesConsumed = 0;
-        while (totalBytesConsumed < inputBytes.Length)
+        while (inputBytes.Length > 0)
         {
-            if (Ascii.IsValid(inputBytes[totalBytesConsumed]))
+            var length = inputBytes.IndexOfAnyExcept(Rfc5987AttrChar);
+            if (length < 0)
+            {
+                length = inputBytes.Length;
+            }
+            int asciiCharCount = Encoding.ASCII.GetChars(inputBytes.Slice(0, length), tempCharBuffer);
```

Comment:
My naive guess would be that this is slower than just casting byte to char and that's why the no-encoding case is slightly slower.

Reply:
Do you want to try re-running the benchmarks with this converted to a simple for-loop, casting and appending each byte?

Reply:
But the "NoEncoding" is actually faster

Reply:
Oops, I got confused.  Any idea why the encoding part is slower?

Reply:
So if I switch this back to a loop, something like:

```csharp
            var asAscii = inputBytes.Slice(0, length);
            while (asAscii.Length > 0)
            {
                builder.Append((char)asAscii[0]);
                asAscii = asAscii.Slice(1);
            }
```

then these would be the results:

```
|                 Method |     Mean |   Error |  StdDev |        Op/s |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------- |---------:|--------:|--------:|------------:|-------:|------:|------:|----------:|
|   FileNameStarEncoding | 217.9 ns | 1.91 ns | 1.69 ns | 4,590,223.2 | 0.0052 |     - |     - |     496 B |
| FileNameStarNoEncoding | 152.4 ns | 1.42 ns | 1.26 ns | 6,561,686.0 | 0.0050 |     - |     - |     464 B |
```

Reply:
I have just pushed an update where I remove the Rune, and now the "encoding" is more-or less the same considering the error. (I suppose it must be a little slower for having to 'allocate' a second char buffer on stack.

Reply:
Yeah, sorry for the confusion about the for-loop - it makes sense that that would be slower.